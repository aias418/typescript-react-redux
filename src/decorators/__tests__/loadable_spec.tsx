// tslint:disable-next-line no-unused-variable
import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'

import loadable from '../loadable'

const TestComponent = () => <div>OK</div>

describe.only('decorators/loadable', () => {

  it('sets friendly .displayName', () => {
    const LoadedTestComponent = loadable(() => false)(TestComponent)
    expect(LoadedTestComponent.displayName)
      .toEqual('Loadable(TestComponent)')
  })

  it('handles loading state', () => {
    const LoadingTestComponent = loadable(() => true)(TestComponent)
    const renderer = TestRenderer.create(<LoadingTestComponent />)
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it('handles loaded state', () => {
    const LoadedTestComponent = loadable(() => false)(TestComponent)
    const renderer = TestRenderer.create(<LoadedTestComponent />)
    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
