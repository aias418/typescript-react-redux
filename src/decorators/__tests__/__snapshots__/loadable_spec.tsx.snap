// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorators/loadable handles loaded state 1`] = `
<div>
  OK
</div>
`;

exports[`decorators/loadable handles loading state 1`] = `
<div>
  Just a moment, please...
</div>
`;
